### PR TITLE
[fuchsia_reload] Find and attach to the first instance of the named app

### PR DESCRIPTION
**Pull request by zanderso**

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/fuchsia_reload.dart [hunk 1]
```diff
@@ -70,24 +71,63 @@ class FuchsiaReloadCommand extends FlutterCommand {
...
+  // Find ports where there is a view isolate with the given name
+  Future<List<int>> _filterPorts(List<int> ports, String isolateFilter) async {
+    final List<int> result = new List<int>();
+    for (int port in ports) {
+      final String addr = "http://$_address:$port";
+      final Uri uri = Uri.parse(addr);
+      final VMService vmService = VMService.connect(uri);
+      await vmService.getVM();
+      await vmService.vm.refreshViews();
+      for (int i = 0; vmService.vm.firstView == null && i < 5; i++) {
```
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/commands/fuchsia_reload.dart]: parens: (vmService.vm.firstView != null)

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 2]
```diff
@@ -219,13 +219,15 @@ abstract class ResidentRunner {
 
     // Refresh the view list.
     await vmService.vm.refreshViews();
-    for (int i = 0; vmService.vm.mainView == null && i < 5; i++) {
+    for (int i = 0; vmService.vm.firstView == null && i < 5; i++) {
```
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/resident_runner.dart]: Ditto about the parens.

This looks like it is duplicated code from above? Please refactor this into a helper function.

File: packages/flutter_tools/lib/src/commands/fuchsia_reload.dart (hunk 1 above)
Comment (zanderso) [inline packages/flutter_tools/lib/src/commands/fuchsia_reload.dart]: Done

File: packages/flutter_tools/lib/src/resident_runner.dart (hunk 2 above)
Comment (zanderso) [inline packages/flutter_tools/lib/src/resident_runner.dart]: Done

**Comment by johnmccutchan** [comment]
LGTM

**Pull request merged**